app/models/user.rb

```diff
@@ -0,0 +1,14 @@
+class User < ApplicationRecord
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-]+(\.[a-z]\d\-]+)*\.[a-z]+\z/i
+  validates :name, presence: true, length: { maximum: Settings.user.name.max_length }
+  validates :email, presence: true, length: { maximum: Settings.user.email.max_length },
+                    format: { with: VALID_EMAIL_REGEX },
+                    uniqueness: { case_sensitive: false }
+  validates :password, presence: true, length: { minimum: Settings.user.password.min_length }
+  before_save :downcase_email
+  private
+  def downcase_email
+    self.email.downcase!
+  end
+  has_secure_password
```

Comment:
method này dùng để làm j cu, giải thích thôi hì

Reply:
@vanvtt-0952 Em xin trả lời là method này dùng để chuyển đổi email về chữ thường, chạy trước khi lưu email người dùng nhập vào database. Hàm downcase! sử dụng trong phương thức này làm thay đổi giá trị của thuộc tính email hiện tại, trả về nil nếu giá trị email người dùng nhập toàn bộ là chữ thường, ở đây em không sử dụng kết hợp với câu lệnh khác nên dùng downcase! cho ngắn gọn.